configure.ac

```diff
@@ -1575,21 +1573,14 @@ dnl                AC_MSG_WARN([Your version of OpenDKIM can not support iSchedu
         dnl httpd needs libmath in a few places
```

Comment:
can you also move this block of comments to the new `-lm` location please. i don't wanna have to dig through history later to figure out wtf they mean, nor do i want their purpose to be lost when reading code at the new location. thanks

Reply:
I do not understand what 
```
dnl XXX really should check for this properly but AC_SEARCH_LIBS/AC_CHECK_LIB
dnl XXX break under -Werror(!)
```
is supposed to mean.  Copying these lines to Makefile.am will make the interpretetation harder.

Reply:
It means we should not be hardcoding `-lm` at all: we should be letting autoconf determine whether it's needed, and if so what it's called.  But we couldn't just do that at the time the comment was written, because the relevant autoconf macros were broken.  I dunno if they're fixed now -- if they are, we can fix `-lm` properly, but that's a separate job.  In the meantime, keep the comment with the `-lm` so the context is preserved.

Reply:
Where does the requirement for libm come from?  libbrotlidec.so has as undefiend symbol log2.  Both versions 1.0.3 and 1.0.7 have libm.so as DT_NEEDED.  sqlite3.pc has in its .pc file `Libs.private: -lz -lm -ldl -lpthread`.

Reply:
lib/bloom.c calls log() and ceil(), so it is a matter passing -lm when linking libcyrus, see https://github.com/cyrusimap/cyrus-imapd/issues/2124.